pinot-query-planner/src/main/java/org/apache/pinot/query/planner/logical/RexExpressionUtils.java

```diff
@@ -61,31 +62,70 @@ static RexExpression handleCast(RexCall rexCall) {
         operands);
   }
 
-  // TODO: Add support for range filter expressions (e.g. a > 0 and a < 30)
   static RexExpression handleSearch(RexCall rexCall) {
     List<RexNode> operands = rexCall.getOperands();
     RexInputRef rexInputRef = (RexInputRef) operands.get(0);
     RexLiteral rexLiteral = (RexLiteral) operands.get(1);
     FieldSpec.DataType dataType = RexExpression.toDataType(rexLiteral.getType());
     Sarg sarg = rexLiteral.getValueAs(Sarg.class);
-    if (sarg.isPoints()) {
-      return new RexExpression.FunctionCall(SqlKind.IN, dataType, SqlKind.IN.name(), toFunctionOperands(rexInputRef,
-          sarg.rangeSet.asRanges(), dataType));
-    } else if (sarg.isComplementedPoints()) {
-      return new RexExpression.FunctionCall(SqlKind.NOT_IN, dataType, SqlKind.NOT_IN.name(),
-          toFunctionOperands(rexInputRef, sarg.rangeSet.complement().asRanges(), dataType));
+    if (sarg.isPoints() || sarg.isComplementedPoints()) {
+      SqlKind kind = sarg.isPoints() ? SqlKind.IN : SqlKind.NOT_IN;
+      Set<Range> rangeSet = kind.equals(SqlKind.IN) ? sarg.rangeSet.asRanges() : sarg.rangeSet.complement().asRanges();
+      List<RexExpression> functionOperands = new ArrayList<>(1 + rangeSet.size());
+      functionOperands.add(RexExpression.toRexExpression(rexInputRef));
+      functionOperands.addAll(rangeSet.stream().map(
+          range -> new RexExpression.Literal(dataType, RexExpression.toRexValue(dataType, range.lowerEndpoint()))
+      ).collect(Collectors.toList()));
+      return new RexExpression.FunctionCall(kind, dataType, kind.name(), functionOperands);
```

Comment:
i wasn't sure if this would be the best way to handle. but let's do it this way. since all range predicates for a single input reference will be lump into one search. let's do
1. all points gets to be in a `IN` search; (only do complemented points if user explicitly said NOT_IN)
2. all continuous ranges be compiled into `OR` search chain. 
link 1 and 2 together with an OR.